app/javascript/Pages/Home.jsx

```diff
@@ -1,37 +1,272 @@
 import React, { Component } from "react";
 import Navbar from "./components/Navbar/Navbar";
 import Footer from "./components/Footer/Footer";
-import { Head, Link } from "@inertiajs/inertia-react";
-import "../stylesheets/Home.css";
-
-class Homepage extends Component {
-  render() {
-    return (
-      <main>
-        <Navbar current_user={this.props.current_user} />
-
-        <img
-          src={require("../images/banner.svg")}
-          alt=""
-          role="presentation"
-          className="hero-img"
-        />
-        <Head title="Welcome" />
-        <p>Hello, welcome to your first Inertia app!</p>
-        <p className="paragraph">This is cool!</p>
-
-        <Link
-          href="/logout"
-          method="delete"
-          headers={window.defaultHeaders}
-          as="button"
+import { Head } from "@inertiajs/inertia-react";
+import { useState, useEffect } from "react";
+import SocialCard from "./components/SocialCard";
+import "../stylesheets/home.scss";
+import ButtonDifficultyFilter from "./components/ButtonDifficultyFilter";
+import CardCategory from "./components/CardCategory";
+import CardPopular from "./components/CardPopular";
+import axios from "axios";
+import ButtonFilter from "./components/ButtonFilter";
+import PropTypes from "prop-types";
+import { useTranslation } from "react-i18next";
+
+function Homepage(props) {
+  const { t, i18n } = useTranslation();
+  const [items, setItems] = useState([]);
+  const [cats, setCat] = useState([]);
+  const [pclasses, setPClasses] = useState([]);
+  const [categories, setCategories] = useState([]);
+  const [categoryFilter, setCategoryFilter] = useState("");
+
+  const fetchClassesDifficulty = async (difficulty) => {
```

Comment:
**issue (blocking)**: currently this function is not async and you don't need to set that (because you are using .then). You can make this async by doing:
```js
const { data } = axios.get(`/api/classes?difficulty=${difficulty}`)
    .catch((error) => {
        console.log(error);
        setItems([]);
    });
setItems(data);
```